### ISSUE TITLE
font declaration error

### ISSUE THREAD

**Issue by Python-Sargent** [remance/Masendor#72]
You seem to have called 'Font' incorrectly:

```
File "C:\Users\user\Documents\Masendor\Masendor-0.7.3.1\engine\game\loading_screen.py", line 12, in loading_screen
    font = pygame.Font(self.ui_font["main_button"], int(60 * self.screen_scale[1]))
AttributeError: module 'pygame' has no attribute 'Font'
```

(full traceback not included)

**Comment by Python-Sargent** [comment]
This may be an error with normal pygame messing with pygame_ce, I use pygame so that's why I have it.

**Comment by remance** [comment]
Hmm yeah those errors are likely to be because of Pygame and pygame-ce being installed both. Your system probably pick pygame for the game. Maybe it is better to stick with Pygame. I assume the package version works fine for you?

**Comment by Python-Sargent** [comment]
No, since I'm using the package version I'm not sure what you saying.

**Comment by Python-Sargent** [comment]
I commented all calls to the blur function, and I get:

```
pickle.dump(new_data, handle)
  File "C:\Users\Samuel\Documents\Masendor\Masendor-0.7.3.1\engine\data\datacacher.py", line 53, in __getstate__
    return (pygame.image.tobytes(self.surface, "RGBA"), self.surface.get_size())
AttributeError: module 'pygame.image' has no attribute 'tobytes'
```

ANother pygame_ce usage...

**Comment by Python-Sargent** [comment]
any way to import pygame_ce in main instead of change all the calls in the various files?

**Comment by remance** [comment]
> any way to import pygame_ce in main instead of change all the calls in the various files?

I am not sure. It should be easy to change requirements and some part of code from using pygame-ce to just pygame. But probably not possible to do specific import like you mentioned.

**Comment by remance** [comment]
Ok the new update change pygame-ce requirement to pygame and fix parts of the error you reported. The game should work now on your end. Let me know how it goes.

**Comment by Python-Sargent** [comment]
Still getting the tobytes error:
```py
File "C:\Users\Samuel\Documents\GitHub\Masendor\engine\data\datacacher.py", line 53, in __getstate__
    return (pygame.image.tobytes(self.surface, "RGBA"), self.surface.get_size())
AttributeError: module 'pygame.image' has no attribute 'tobytes'
```

**Comment by Python-Sargent** [comment]
I also had to fix the missing files error again, but it was pretty easy to do.

From what it looks like all the files are there but aren't named in a way that the engine is looking for.

**Comment by remance** [comment]
Hmm weird, I never get this error before. By the way you probably should uninstall pygame-ce now in case there are conflicts with normal pygame. Also for reference, what OS are you using?

**Comment by Python-Sargent** [comment]
I'm using Windows 10 because my PC is broke. It's an old Lenovo laptop.

**Comment by Python-Sargent** [comment]
Another bug here, must be a nil variable or something:

```py
File "C:\Users\user\Documents\GitHub\Masendor\engine\game\create_troop_sprite_pool.py", line 308, in create_sprite
    animation_sprite_pool[sprite_id] = load_pickle_with_surfaces(cache_file_path)
  File "C:\Users\user\Documents\GitHub\Masendor\engine\data\datacacher.py", line 36, in load_pickle_with_surfaces
    data = pickle.load(handle)
EOFError: Ran out of input
```

**Comment by remance** [comment]
> ```python
> EOFError: Ran out of input
> ```

Seem to be an issue of empty cache file. Maybe the game crashed or closed while the cache data being stored. A quick fix would be to remove cache folder in the game folder. But thanks for mentioning this, need to add a function to catch this error in game code so the game can recreate cache file again.

**Comment by Python-Sargent** [comment]
Deleting cache doesn't do anything. Also, I don't think it was because of a crash corruption, seeing as I've never gotten far enough to totally load anything that should be cached.

**Comment by remance** [comment]
That is weird because  "data = pickle.load(handle)" is for cache file loading. 

> Still getting the tobytes error:
> 
> ```python
> File "C:\Users\user\Documents\GitHub\Masendor\engine\data\datacacher.py", line 53, in __getstate__
>     return (pygame.image.tobytes(self.surface, "RGBA"), self.surface.get_size())
> AttributeError: module 'pygame.image' has no attribute 'tobytes'
> ```

Do you still get this error also? Can you post your python package? Maybe it is issue of incompatible pygame version.

**Comment by Python-Sargent** [comment]
I'm using everything latest.

I'm not sure what else is causing it to crash.

```shell
Python 3.10.4
```

```shell
Name: pygame
Version: 2.5.0
Summary: Python Game Development
Home-page: https://www.pygame.org
Author: A community project.
Author-email: pygame@pygame.org
License: LGPL
Location: c:\users\user\appdata\local\programs\python\python310\lib\site-packages
Requires:
Required-by: pgzero
```

**Comment by remance** [comment]
The error makes no sense because tobytes is a function for pygame.image. Maybe worth making new clean virtual environment?

https://pyga.me/docs/ref/image.html#pygame.image.tobytes

**Comment by Python-Sargent** [comment]
pardon my stupidity, but why do people put 'extra' dots (.) in URLs, is that a european thing or somthing?

Also, I using vanilla Python 3, so it's not with a virtualenv, I'm using the version I installed on the system.

**Comment by remance** [comment]
Dunno about URL extra dots to be honest. Probably related to domain or specific HTML structure or something. 

I would recommend you setup virtualenv for each project or maybe, one for your working and one for other purposes. That way it will be much easier to avoid package issues and a bit easier to organise.

**Comment by Python-Sargent** [comment]
Same error with virtualenv.

Not sure what's wrong.

Note: I followed your advice and this is a local virtualenv, in the Masendor directory.

**Comment by Python-Sargent** [comment]
I reset it and changed the requirements.txt to show `pygame_ce~=2.3.1`.

Re-ran the pip install in a new virtualenv, and crashed like before.

**Comment by remance** [comment]
So it crash for both pygame-ce and normal pygame? Both same error? I assume it the AttributeError: module 'pygame.image' has no attribute 'tobytes'

**Comment by Python-Sargent** [comment]
Yes, sorry for not being clear.

**Comment by remance** [comment]
Very weird for sure, wonder what the issue stem from.
